server/map_rpc_server.go

```diff
@@ -112,8 +112,21 @@ func (t *TrillianMapServer) Init(ctx context.Context, mapID int64) error {
 // GetLeaves implements the GetLeaves RPC method.  Each requested index will
 // return an inclusion proof to either the leaf, or nil if the leaf does not
 // exist.
+// TODO(phad): this will soon change to only request the most recent version
+// of the specified leaf indices.
 func (t *TrillianMapServer) GetLeaves(ctx context.Context, req *trillian.GetMapLeavesRequest) (*trillian.GetMapLeavesResponse, error) {
-	mapID := req.MapId
+	return t.getLeavesByRevision(ctx, req.MapId, req.Index, req.Revision)
```

Comment:
Should `GetLeaves` call `GetLeavesByRevision` directly?  
It would benefit from the additional input validation in `GetLeavesByRevision`